controller/api/public/edges.go

```diff
@@ -96,14 +97,22 @@ func processEdgeMetrics(inbound, outbound model.Vector, resourceType, selectedNa
 	for _, sample := range inbound {
 		// skip inbound results without a clientID because we cannot construct edge
 		// information
-		if _, ok := sample.Metric[model.LabelName("client_id")]; ok {
-			key := sample.Metric[model.LabelName(resourceReplacementInbound)]
+		if clientID, ok := sample.Metric[model.LabelName("client_id")]; ok {
+			dstResource := string(sample.Metric[model.LabelName(resourceReplacementInbound)])
+
+			// format of clientId is id.namespace.serviceaccount.cluster.local
+			clientIDSlice := strings.Split(string(clientID), ".")
+			srcNs := clientIDSlice[1]
+			key := model.LabelValue(dstResource + srcNs)
```

Comment:
Let's use a delimiter here to separate the name and the namespace, so there's no ambiguity as to which part of the string is the name, and which part is the namespace. I recommend using a ".", since that's no a valid character for names or namespaces. How about something like:

```suggestion
			key := model.LabelValue(fmt.Sprintf("%s.%s", dstResource, srcNs))
```

Reply:
👍 Great idea, this will make it easier to read if debugging later.